railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -305,6 +305,17 @@ def delete_application_layout_file_if_api_option
         end
       end
 
+      def delete_public_files_if_api_option
+        if options[:api]
+          remove_file 'public/404.html'
+          remove_file 'public/422.html'
+          remove_file 'public/500.html'
+          remove_file 'public/apple-touch-icon-precomposed.png'
+          remove_file 'public/apple-touch-icon.png'
+          remove_file 'public/favicon.ico'
```

Comment:
It is possible to suppress for html files not for images. Also we are removing all assets, helpers and other unwanted stuff for API apps. So this seems fine. Infact we can remove all the conditionals from assets, helpers related files as they are of no use because we just remove them anyways later.